src/tunacode/infrastructure/cache/manager.py

```
@@ -0,0 +1,178 @@
+    """Protocol for cache invalidation strategies."""
+
+    def is_stale(self, key: str, metadata: dict[str, Any]) -> bool: ...
+
+
+class ManualStrategy:
+    """Invalidation only via explicit clear_cache() calls.
+
+    Used for: agents, settings, models registry, xml prompts, ripgrep utils.
+    """
+
+    def is_stale(self, key: str, metadata: dict[str, Any]) -> bool:
```

Ruff (0.14.14)

[warning] Unused method argument: `key`

(ARG002)